packages/components/checkbox/src/Checkbox.tsx

```diff
@@ -22,41 +24,38 @@ export const Checkbox = forwardRef<HTMLButtonElement, CheckboxProps>(
     },
     forwardedRef
   ) => {
+    const innerId = useId()
+    const innerLabelId = useId()
+
     const field = useFormFieldState()
     const group = useCheckboxGroup()
+
     const rootRef = useRef<HTMLButtonElement | undefined>()
     const ref = useMergeRefs(forwardedRef, rootRef)
 
-    const name = field.name ?? group.name
-    const isRequired = field.isRequired ?? group.isRequired
-    const isInvalid = field.isInvalid ?? group.isInvalid
-    const isFieldEnclosed = field.id !== group.id
-    const id = isFieldEnclosed ? field.id : undefined
-    const description = isFieldEnclosed ? field.description : undefined
-
-    const intent = (() => {
-      if (isInvalid) return 'error'
-
-      return intentProp ?? group.intent
-    })()
-
-    const checked = group.value && value ? group.value.includes(value) : checkedProp
+    const checked = value ? group.value?.includes(value) : checkedProp
 
     const handleCheckedChange = (isChecked: boolean) => {
-      if (onCheckedChange) onCheckedChange(isChecked)
+      onCheckedChange?.(isChecked)
 
       const rootRefValue = rootRef.current?.value
-
-      if (!rootRefValue || !group.onCheckedChange) return
-
-      group.onCheckedChange(isChecked, rootRefValue)
+      rootRefValue && group.onCheckedChange?.(isChecked, rootRefValue)
```

Comment:
I think the other way around was more legible. As a general rule I never inline `if` statements like you are doing